src/GafferImage/Display.cpp

```diff
@@ -268,43 +301,56 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			Display::driverCreatedSignal()( driver.get(), parameters.get() );
 		}
 
-		ConstFloatVectorDataPtr getTile( const V2i &tileOrigin, size_t channelIndex )
+		struct Tile
 		{
-			V2i tileIndex = tileOrigin / ImagePlug::tileSize();
-
-			if(
-				tileIndex.x < m_tiles.index_bases()[0] ||
-				tileIndex.x >= (int)(m_tiles.index_bases()[0] + m_tiles.shape()[0] ) ||
-				tileIndex.y < m_tiles.index_bases()[1] ||
-				tileIndex.y >= (int)(m_tiles.index_bases()[1] + m_tiles.shape()[1] )
-			)
+			Tile(): backBufferData( ImagePlug::blackTile()->copy() ), backBuffer( backBufferData->writable() ), dirty( false ), cachedTile( ImagePlug::blackTile() ), cachedForDataCount( 0 ), updateCount( 0 )
 			{
-				// outside data window
-				return nullptr;
 			}
 
-			tbb::spin_rw_mutex::scoped_lock tileLock( m_tileMutex, false /* read */ );
+			Tile( const Tile& other ) : Tile()
+			{
+				*this = other;
+			}
 
-			ConstFloatVectorDataPtr result = m_tiles[tileIndex.x][tileIndex.y][channelIndex];
-			if( !result )
+			Tile& operator=( const Tile& other )
 			{
-				result = ImagePlug::blackTile();
+				memcpy( &backBuffer[0], &other.backBuffer[0], backBuffer.size() * sizeof( float ) );
+				dirty = (bool)other.dirty;
+				cachedTile = other.cachedTile;
+				cachedForDataCount = other.cachedForDataCount;
```

Comment:
I seem to remember you saying this was a bug previously, and we needed to reset the `cachedForDataCount`. Did I misremember?

Reply:
I have brought back the previous commit addressing this: "Display : Possible solution to setDriver".  I recall you not being entirely happy with this, but I'm having a hard time finding your comments, maybe we talked about it verbally?  If we're sticking with this approach for the moment, do you have suggestions for comments?

Reply:
Yeah, I think we chatted about this verbally. I think we thought that this might be a slightly more explicit way of doing the same thing :

- Rename the `setDriver()` `copy` parameter to `snapshot`, to make it much clearer that we're expecting a static copy that is no longer attached to incoming buckets.
- Ditch the Tile copy constructor, and use a more explicit `snapshot()` method to match the above.
- A snapshotted Tile could then be `!dirty`, with `cachedTile` already set up and `cachedForDataCount == 0`, or perhaps a named `g_snapshotDataCount` value.
- `setDriver()` could then reset channelDataCount to `g_snapshotDataCount` to make the link clear.

I'm not sure this is worth the effort right now though. I've made a couple of suggestions on the comments in your commit instead, which I hope help clarify what we're doing.